instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,311 @@
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSession()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isActiveGuidanceSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testOneBillingSessionSentForOneSecondLongPausedAndResumedActiveGuidanceTrip() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        val sessionLengthMillis = TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS)
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        delay(sessionLengthMillis / 2)
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        delay(sessionLengthMillis / 2)
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isActiveGuidanceSession)
+
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testOneBillingSessionSentForEmptyPausedAndResumedActiveGuidanceTrip() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isActiveGuidanceSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForOneSecondLongFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
```

Comment:
I think it would be better to invoke `stopTripSession` here. You start FD with `startTripSession` and stop with `stopTripSession`. `onDestroy` does more, it also doesn't allow you to use `MapboxNavigation` instance anymore. I understand that you don't need it here, but just stopping the session would be a "cleaner" way IMO.

Reply:
Calling only `stopTripSession()` is not enough because it only pauses current session https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/BillingController.kt#L216

Billing session is stopped only when [onDestroy()](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/BillingController.kt#L305) called or when another session [starts](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/BillingController.kt#L342).

That's how it works now. We shouldn't change behaviour in this PR but should discuss whether this is a desired behaviour. cc @VysotskiVadim